uni/lib/controller/fetchers/session_dependant_fetcher.dart

```diff
@@ -1,5 +1,10 @@
 import 'package:uni/model/entities/session.dart';
 
+// TODO(luisd): apparently, it's not a good practice defining an abstract class
+//  with just one function, it's better to typedef the function, we can disable
+//  rule if we want a more java-like experience.
+
+//ignore: one_member_abstracts
```

Comment:
I'm not a huge fan of that rule either, should we disable it manually?

Reply:
Well, it makes more sense for me to have the abstract class on one member abstracts, since makes extensibility easier on the future, i'll disable it then